### ISSUE TITLE
GPU Performance

### ISSUE THREAD

**Issue by jake1271** [Const-me/Whisper#85]
I have a few GPUs from my two mining rigs, thought I'd share my findings.

Setup: i3-13100F cpu  (single core Geekbench 6 is 2300)
GPUs: 2080 Ti 11gb 352-bit bus, 3070 8gb 256-bit bus, 3060 Ti 8gb 256-bit bus
RAM: 16gb

Transcribing an 8 min English audio clip:
1. Using main 16x pcie slot.  Both the 2080 Ti and 3060 Ti will transcribe an 8min audio using medium model in about 45 seconds. 
2. Using a pcie 1x slot extender. 3060 Ti takes 2min 25 secs. The 3070 is about few seconds faster at 2m 17secs.

Using the OpenAI Whisper with Pytorch support for GPU, there is little difference between using the pcie 16x or 1x slot. The 3070 will do the same 8min transcription in 1m 50sec on 16x, loses a few seconds due to being on the pcie 1x slot but not much difference.  On a side note the M1 Max using Whisper.cpp will do the 8m transcription in a similar 1m 45sec, so M1 Max cpu = 3070 gpu.

Not sure why the 2080 Ti and 3060 Ti are so close in performance when the 2080 Ti is 60% faster with FP16, perhaps CPU bottle necking? CPU utilization is only around 20%, but something seems to be bottle necking the GPUs.

Conclusion, the Const-me implementation is blazing fast on the 16x slot and seems like there's more GPU performance to unlock and some possible optimizations for slower slots (1x).

**Comment by emcodem** [comment]
If you share your test audio file, i can add A5000 and A3000 mobile speeds :D

**Comment by jake1271** [comment]
Sure here ya go, it's a tutorial on installing Whisper.
https://user-images.githubusercontent.com/51808736/231159408-ddfce1a5-a633-46c0-9d44-cbd9af14f531.mp4

And for those that are wondering what the heck a pcie 1x slot extender is, it's the sled the GPUs are sitting on in the pic, mainly used on mining rigs to hookup multiple gpus.

![20230411_192637_resized](https://user-images.githubusercontent.com/51808736/231172946-e3f1a75e-ebbb-4f2a-afc6-8a03f7703af7.jpg)

**Comment by emcodem** [comment]
Thanks, here the results from Cards that i have easy access to, both Professional versions.

**Model Large (V2)**
Nvidia RTX A3000 Laptop GPU/Core i9 
RunComplete     121.833 seconds

RTX Quadro A5000 (Desktop) / Core i9 9900K
RunComplete     87.1558 seconds

**Model Medium**
Nvidia RTX A3000 Laptop GPU/Core i9 
RunComplete     77.5319 seconds

RTX Quadro A5000 (Desktop) / Core i9 9900K
RunComplete     49.7725 seconds


While operating, the cpu (on laptop and desktop, independent of the used model) looks like this. Which indicates a "slight" CPU Speed Bottleneck on my machine but nothing worrying:

<img width="527" alt="cpu" src="https://user-images.githubusercontent.com/53784010/231462384-25f854cf-e376-43bc-bb00-031c8b6751db.png">

**Comment by emcodem** [comment]
Newest tests show that the original project whisper cpp actually can run faster than my A5000 above by about 20% while utilizing far less power on apple M1 ultra ^^

**Comment by NikosDi** [comment]
Intel Arc A380 using WhisperDesktop v1.12

medium.en model:   93 sec
large-v1 model:        171 sec

@emcodem 

What app did you use for large v2 model ?
Because I think WhisperDesktop is incompatible.

**Comment by RickArcher108** [comment]
I use Whisper Desktop and Whisperer. Both of them work fine with the large model on my PC. I have a 12gb GPU.

Rick Archer
Buddha at the Gas Pump
https://batgap.com
***@***.***

**Comment by NikosDi** [comment]
Yes, I tried both and work fine too with my A380 - large-v1 and large (v2)
Large (v2) is a little slower, but probably more accurate.

The reason I asked is because a lot of people were mentioning in the open issues of this app that large (v2) is buggy and doesn't work OK with WhisperDesktop, not because of VRAM requirements which are the same with v1.

But it seems working fine.

**Comment by emcodem** [comment]
@NikosDi i never tried large-v1 but i wouldn't expect to see much difference between v1 and v2, not in terms of VRAM usage nor in accuracy speed and or "errors". If you refer to https://github.com/Const-me/Whisper/issues/166 with your assumption about V2 not working in whisperdesktop, note that the problem there was that the asker used a totally wrong model format (non ggml)

**Comment by albino1** [comment]
I assume he's talking about this thread using the president audio test example where large-v2 is much worse with the Const-me Whisper:

https://github.com/Const-me/Whisper/issues/61

Maybe it's been resolved since originally posted, but I haven't done any recent testing, and the models haven't been updated to resolve any potential conversion issues anyway if that's the root issue like some have suggested.

**Comment by emcodem** [comment]
@albino1  Funny enough i am also under the impression that the simple greedy search of const-me version is generally equal if not better than beam search (which other projects tend to default to). I worked into some documents and the source code of the other projects, leaving me under the impression that the others are kind of over engineering everything, trying to apply lessons learned from chat  (GPT) models(but these lessons do not completely apply to the whisper usecase)  and especially trying to overcome forever repeated output by applying a lot of completely worthless logic ^^

**Comment by Whisper-Padawan** [comment]
> Intel Arc A380 using WhisperDesktop v1.12
> 
> medium.en model: 93 sec **large-v1 model: 171 sec**


Hey bro, i've test this mp4 file with my 5600G Vega: 404s.
Your GPU is pretty fast.
